deploy.sh

```diff
@@ -10,15 +10,17 @@ sed -i.bak \
 rm deploy/kubernetes/storage.yaml.bak
 
 sed -i.bak \
-    -e "s/NOTEBOOK_VERSION_VALUE/${NOTEBOOK_VERSION}/g" \
+    -e "s/BASE_STACK_VALUE/${BASE_STACK}/g" \
+    -e "s/STACKS_VALUE/${STACKS}/g" \
+    -e "s/STACKS_NAMES_VALUE/${STACKS_NAMES}/g" \
     -e "s/STORAGE_CLASS_VALUE/${STORAGE_CLASS}/g" \
     -e "s/STORAGE_PER_USER_VALUE/${STORAGE_PER_USER}/g" \
     -e "s/WIPP_STORAGE_PVC_VALUE/${WIPP_STORAGE_PVC}/g" \
     -e "s|WIPP_UI_VALUE|${WIPP_UI}|g" \
     -e "s|WIPP_API_INTERNAL_VALUE|${WIPP_API_INTERNAL}|g" \
     -e "s|WIPP_NOTEBOOKS_PATH_VALUE|${WIPP_NOTEBOOKS_PATH}|g" \
-    deploy/kubernetes/jupyterhub-configs.yaml
-rm deploy/kubernetes/jupyterhub-configs.yaml.bak
+    deploy/kubernetes/jupyterhub-config.py
+rm deploy/kubernetes/jupyterhub-config.py.bak
 
 CONFIG_HASH=$(shasum deploy/kubernetes/jupyterhub-configs.yaml | cut -d ' ' -f 1 | tr -d '\n')
```

Comment:
We hash the config yaml and add it as an annotation in the deployment to trigger automated redeployment of JupyterHub pods when the config map changes. Can you separate out the Spawner class and cull-idle-server files as separate python files, but leave strictly the configurations in the previous .yaml format?

Reply:
Done